tests/e2e/specs/auth-flow-editor.test.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activatePlugin,
+	loginUser,
+	createURL,
+	visitAdminPage,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	setAuthToken,
+	setClientConfig,
+	setSearchConsoleProperty,
+	setSiteVerification,
+} from '../utils';
+
+function requestHandler( request ) {
+	if ( ! request._allowInterception ) {
+
+		// prevent errors for requests that happen after interception is disabled
+		return;
+	} else if ( request.url().startsWith( 'https://accounts.google.com/o/oauth2/auth' ) ) {
+		request.respond( {
+			status: 302,
+			headers: {
+				location: createURL( '/', 'oauth2callback=1&code=valid-test-code' )
+			}
+		} );
+	} else {
+		request.continue();
+	}
+}
+
+describe( 'the set up flow for an editor', () => {
+
+	beforeAll( async() => {
+		await page.setRequestInterception( true );
+		page.on( 'request', requestHandler );
+	} );
+
+	beforeEach( async() => {
+		await activatePlugin( 'e2e-tests-oauth-callback-plugin' );
+		await setClientConfig();
+		await setAuthToken();
+		await setSiteVerification();
+		await setSearchConsoleProperty();
+	} );
+
+	afterEach( async() => {
+
+		// Restore the default/admin user
+		// (switchToAdmin will not work as it is not aware of the current user)
```

Comment:
Meaning that `WP_ADMIN_USER` there is not defined?

Reply:
Meaning that the `switchUserTo*` utilities are env-var based and are not aware of who the currently logged in user is. In our case, these users are the the same (admin) so these utilities actually [do nothing](https://github.com/WordPress/gutenberg/blob/be7db4f6c98857bae01180b07db0e8a2ec18f6d1/packages/e2e-test-utils/src/switch-user-to-admin.js).